techcity/groups/management/commands/populate_groups.py

```diff
@@ -0,0 +1,54 @@
+import requests
+from django.core.management.base import BaseCommand
+from requests.adapters import HTTPAdapter
+from urllib3.util import Retry
+
+from techcity.groups.models import Group
+
+
+class Command(BaseCommand):
+    help = "Fetch and upsert groups from the deployed API"
+
+    def handle(self, *args, **kwargs):
+        groups_endpoint = "https://community.techfrederick.org/api/v1/groups"
+        retries = Retry(
+            total=3,
+            allowed_methods={"GET"},
+            status_forcelist=[502, 503, 504],
+            backoff_factor=0.1,
+        )
+        session = requests.Session()
+        session.mount("https://", HTTPAdapter(max_retries=retries))
+
+        try:
+            groups_response = session.get(groups_endpoint)
+            groups_response.raise_for_status()
+        except requests.exceptions.RequestException as e:
+            self.stdout.write(
+                self.style.NOTICE(f"An error occurred: {e}, check {groups_endpoint}")
+            )
+            return
+
+        groups = groups_response.json()
+        if not groups:
```

Comment:
Can we flip this logic so there isn't a negative if clause (i.e., `if groups` instead of `if not groups`)? Or, if you want to keep in the guard clause style, you could drop the `else:` and outdent the code that is currently in the `else:` clause.